tests/v2/validation/rbac/ingress/ingress.go

```diff
@@ -0,0 +1,394 @@
+	}
+	if err := steveClient.SteveType("v1.namespace").Delete(nsObj); err != nil {
+		return fmt.Errorf("failed to delete namespace %s: %v", namespaceName, err)
+	}
+
+	waitErr = wait.Poll(2*time.Second, 30*time.Second, func() (bool, error) {
+		_, err := steveClient.SteveType("v1.namespace").ByID(namespaceName)
+		if err != nil {
+			return true, nil
+		}
+		log.Infof("Waiting for namespace %s to be removed...", namespaceName)
+		return false, nil
+	})
+	if waitErr != nil {
+		return fmt.Errorf("timeout waiting for namespace %s to be removed", namespaceName)
+	}
+
+	log.Infof("Cleanup completed successfully for namespace: %s", namespaceName)
+	return nil
+}
+
+// setupTest creates necessary test resources including project, namespace and downstream client.
+func setupTest(client *rancher.Client, clusterID string, name string) (*v1.Client, *management.Project, *corev1.Namespace, error) {
+	log.Infof("Setting up test resources for: %s", name)
+
+	project, namespace, err := projects.CreateProjectAndNamespace(client, clusterID)
+	if err != nil {
+		log.Errorf("Failed to create project and namespace for test %s: %v", name, err)
+		return nil, nil, nil, fmt.Errorf("project and namespace creation failed: %v", err)
+	}
+	log.Infof("Created project %s and namespace %s", project.Name, namespace.Name)
+
+	steveClient, err := client.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		log.Errorf("Failed to get downstream client for test %s: %v", name, err)
+		return nil, nil, nil, fmt.Errorf("downstream client creation failed: %v", err)
+	}
+
+	return steveClient, project, namespace, nil
+}
+
+// createWorkload creates a deployment workload with the specified parameters.
+func createWorkload(client *rancher.Client, clusterID, namespace string, replicas int) (*v2.Deployment, error) {
+	log.Infof("Creating workload in namespace %s with %d replicas", namespace, replicas)
+	deployment, err := deployment.CreateDeployment(client, clusterID, namespace, replicas, "nginx", "", false, false, true, false)
+	if err != nil {
+		log.Errorf("Failed to create workload: %v", err)
+		return nil, fmt.Errorf("workload creation failed: %v", err)
+	}
+	log.Infof("Successfully created workload: %s", deployment.Name)
+	return deployment, nil
+}
+
+// generateUniqueHost generates a unique hostname for ingress testing.
+func generateUniqueHost(t *testing.T, prefix string) string {
+	testName := strings.ToLower(strings.ReplaceAll(t.Name(), "/", "-"))
+	randomStr := namegen.AppendRandomString("")
+	if prefix == "" {
+		prefix = "test"
+	}
+	host := fmt.Sprintf("%s-%s-%s.example.com", prefix, testName, randomStr)
+	log.Infof("Generated unique host: %s", host)
+	return host
+}
+
+// findIngressSchema finds and returns the ingress schema from the schema collection.
+func findIngressSchema(schemas *v1.SteveCollection) *v1.SteveAPIObject {
+	log.Info("Searching for ingress schema")
+	for _, schema := range schemas.Data {
+		if schema.ID == "networking.k8s.io.ingress" {
```

Comment:
You could use the constant that's already available here: https://github.com/rancher/shepherd/blob/98e341c77d0b657933e25df39e1f9bb33c4870d6/extensions/ingresses/ingresses.go#L22C2-L22C18